eclipsesource/theia
# Add code response part renderer

#### What it does

Close https://github.com/eclipsesource/osweek-2024/issues/19
Part of https://github.com/eclipsesource/osweek-2024/issues/54

Adds a renderer for a code block using the monaco editor.

![image](https://github.com/user-attachments/assets/4c77ed4a-59f1-445b-b9e4-0761a5932a75)

Implements:
- Copy
- Insert at cursor
- Monaco Editor
- Navigating to the location of the file (if provided)

The parser needs improvement.

#### How to test

0. Install the TypeScript Extension for Theia (for syntax highlighting)
1. packages/ai-chat/src/common/mock-code-chat-agent.ts Replace the URI.
2. Send a message to the bot
3. Click on the buttons / see highlighted code

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
